### DISCUSSION TITLE
Pnp Core "Getting a single file"

### DISCUSSION THREAD

**Discussion by matjazreberc** [pnp/pnpcore#1139] [General] [answered]
Hello,

I'am traying to use, [https://pnp.github.io/pnpcore/using-the-sdk/files-intro.html#getting-a-single-file](url)
`Guid fileId = Guid.Parse("{D61B52DE-2ED1-41E2-B3FC-360E7286B0F9}");

// Get a reference to the file
IFile testDocument = await context.Web.GetFileByIdAsync(fileId);`

but I get error, 
**"'IWeb' does not contain a definition for 'GetFileByIdAsync' and no accessible extension method 'GetFileByIdAsync' accepting a first argument of type 'IWeb' could be found (are you missing a using directive or an assembly reference?)"**

What am I doing wrong?

Thank you, Matjaz

**Comment by jansenbe** [comment, marked as answer]
@matjazreberc : that method was added after 1.8 shipped, you need to use our nightly builds. We plan to ship 1.9 around the end of the month.

**Reply by matjazreberc** [reply]
Ok.
Thank you.
Best regards, Matjaz
